flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/AvroRowDeserializationSchema.java

```diff
@@ -340,10 +357,22 @@ private Time convertToTime(Object object, @Nullable JodaConverter jodaConverter)
 		return new Time(millis - LOCAL_TZ.getOffset(millis));
 	}
 
-	private Timestamp convertToTimestamp(Object object) {
+	private Timestamp convertToTimestamp(Object object, @Nullable JodaConverter jodaConverter, boolean isMicros) {
 		final long millis;
 		if (object instanceof Long) {
-			millis = (Long) object;
+			if (isMicros) {
+				long micros = (Long) object;
+				Instant instant = Instant.ofEpochSecond(0)
```

Comment:
maybe to much optimization but should we just do the long arithmetic ourselves here and below? We are creating a lot of objects for the hot path.